src/metabase/util/honey_sql_2.clj

```diff
@@ -141,6 +141,14 @@
               :when     (some? component)]
           (u/qualified-name component)))])
 
+(mu/defn identifier->components :- [:sequential string?]
```

Comment:
Can you explain why the return type is `[:sequential string?]`?

It looks like from this schema the components can be either keywords, non-blank strings, or identifiers
https://github.com/metabase/metabase/blob/f825f1c13d37486ac7c746343df140dda65ccf38/src/metabase/util/honey_sql_2.clj#L134

Reply:
good question. if you look into the implementation of `identifier` for honeysql 2, all args are unpacked and converted to strings.

https://github.com/metabase/metabase/blob/992f2c27bf0d4ed83b947bcaaae394fd79ebc181/src/metabase/util/honey_sql_2.clj#L142

Also, the last line of that function's doc says the same thing.